### PR TITLE
Reorganize UI test JS utils into Ide and Project + add Ide.awaitNoBackgroundTask()

### PR DESCRIPTION
**Pull request by PawelLipski**
Needed for TDD for #424

### CODE REVIEW

File: uiTests/src/test/resources/scenarios-and-utils.rhino.js [hunk 1]
```diff
@@ -11,113 +12,135 @@ importClass(com.intellij.openapi.actionSystem.AnActionEvent);
...
 }
 
-// Do not run in EDT.
-function configureIde() {
-  const settings = GeneralSettings.getInstance();
-  settings.setConfirmExit(false);
-  settings.setShowTipsOnStartup(false);
+// Do not run any of the methods in EDT.
+function Ide() {
+  this.configure = function () {
```
Comment (PawelLipski) [inline uiTests/src/test/resources/scenarios-and-utils.rhino.js:30]: A prime example of OOP made from _plywood and cardboard_ :trollface:

File: uiTests/src/test/resources/scenarios-and-utils.rhino.js [hunk 2]
```diff
@@ -11,113 +12,135 @@ importClass(com.intellij.openapi.actionSystem.AnActionEvent);
...
+        return getMethod(clazz.getSuperclass(), methodName);
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  this.awaitNoBackgroundTask = function () {
+    const progressManager = ProgressManager.getInstance();
+    const method = getMethod(progressManager.getClass(), 'getCurrentIndicators');
```
Comment (PawelLipski) [inline uiTests/src/test/resources/scenarios-and-utils.rhino.js:58]: Apparently no public API gives access to ongoing background tasks :frowning_face:

File: uiTests/src/test/resources/ide.rhino.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    ProjectUtil.openOrImport(path, /* projectToClose */ null, /* forceOpenInNewFrame */ false);
+  };
+
+  this.soleOpenedProject = function () {
+    return new Project(ProjectUtil.getOpenProjects()[0]);
+  }
+
+  this.awaitNoBackgroundTask = function () {
+    const progressManager = ProgressManager.getInstance();
+    const method = getMethod(progressManager.getClass(), 'getCurrentIndicators');
```
Comment (PawelLipski) [inline uiTests/src/test/resources/ide.rhino.js:29]: Apparently no public API gives access to the list of all ongoing background tasks :frowning_face:

File: uiTests/src/test/resources/ide.rhino.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+importClass(com.intellij.ide.GeneralSettings);
+importClass(com.intellij.ide.impl.ProjectUtil);
+importClass(com.intellij.openapi.application.ApplicationManager);
+importClass(com.intellij.openapi.application.ModalityState);
+importClass(com.intellij.openapi.progress.ProgressManager);
+importClass(com.intellij.ui.GuiUtils);
+
+// Do not run any of the methods in EDT.
+function Ide() {
+  this.configure = function () {
```
Comment (PawelLipski) [inline uiTests/src/test/resources/ide.rhino.js:13]: A prime example of OOP made of _plywood and cardboard_ :trollface:

Comment (mkondratek) [reply uiTests/src/test/resources/ide.rhino.js:13]: :heart_decoration:

File: uiTests/src/test/resources/project.rhino.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  this.openTab = function () {
+    const toolWindowManager = ToolWindowManager.getInstance(underlyingProject);
+    let toolWindow;
+    do {
+      toolWindow = toolWindowManager.getToolWindow(ToolWindowId.VCS);
+      sleep();
+    } while (toolWindow == null);
+
+    // The test is (obviously) not run on UI thread,
+    // so `runOrInvokeAndWait` really means `enqueue onto UI thread and wait until complete`.
```
Comment (mkondratek) [inline uiTests/src/test/resources/project.rhino.js:21]: `onto the UI thread` :question: :gammar-lover:

Comment (PawelLipski) [reply uiTests/src/test/resources/project.rhino.js:21]: Fixed everywhere in the project

**Pull request merged**